```
=================================== FAILURES ===================================
_________________ test_inline_plugin_wait_for_log_no_selfmatch _________________
[gw0] linux -- Python 3.10.20 /home/runner/work/lightning/lightning/.venv/bin/python

node_factory = <pyln.testing.utils.NodeFactory object at 0x7f8fe41b0fa0>

    def test_inline_plugin_wait_for_log_no_selfmatch(node_factory):
        """On inline-plugin nodes the test process's logging is forwarded into
        the node's log.  wait_for_log()'s own 'Waiting for [pattern]'
        announcement embeds the pattern, so with test logging at DEBUG it used
        to land in the scanned log and match itself, reducing the wait to a
        no-op (#9343).  A pattern that never appears must genuinely time out.
        """
        def setup(plugin):
            @plugin.method('inline_ping')
            def inline_ping(plugin):
                logging.info("AUTHOR_LOG_MARKER_9343")
                return {'pong': True}
    
>       l1 = node_factory.get_node(inline_plugin=setup)

tests/test_plugin.py:1817: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
contrib/pyln-testing/pyln/testing/utils.py:1898: in get_node
    _inline_plugin(node, inline_plugin)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = <fixtures.LightningNode object at 0x7f8ffe6e3fd0>
setup_fn = <function test_inline_plugin_wait_for_log_no_selfmatch.<locals>.setup at 0x7f8fff4ca560>

    def _inline_plugin(node, setup_fn):
        """Set up an inline plugin serve thread for a not-yet-started node.
    
        Normally called via get_node(inline_plugin=setup_fn).  The plugin's cwd
        (set by lightningd) is node.daemon.lightning_dir/TEST_NETWORK/, which is
        where the shim looks for inline-plugin.sock.
    
        Example::
    
            def setup(plugin):
                @plugin.method('greet')
                def greet(name, plugin):
                    return {'message': f'hello {name}'}
    
            l1 = node_factory.get_node(inline_plugin=setup)
            assert l1.rpc.greet('world') == {'message': 'hello world'}
        """
        sock_path = os.path.join(node.daemon.lightning_dir, TEST_NETWORK, 'inline-plugin.sock')
        srv = socket.socket(socket.AF_UNIX)
>       srv.bind(sock_path)
E       OSError: AF_UNIX path too long

contrib/pyln-testing/pyln/testing/utils.py:2055: OSError
----------------------------- Captured stdout call -----------------------------
{'run_id': 338097548104253440, 'github_repository': 'ElementsProject/lightning', 'github_sha': '751601e833243e0791cdaee4ca3cf428123d3f8b', 'github_ref': 'refs/pull/9344/merge', 'github_ref_name': 'HEAD', 'github_run_id': 29874630952, 'github_head_ref': 'fix-inline-plugin-log-selfmatch', 'github_run_number': 17275, 'github_base_ref': 'master', 'github_run_attempt': '1', 'testname': 'test_inline_plugin_wait_for_log_no_selfmatch', 'start_time': 1784678189, 'end_time': 1784678189, 'outcome': 'fail'}
----- generated xml file: /home/runner/work/lightning/lightning/report.xml -----
============================= slowest 10 durations =============================
475.22s call     test_connection.py::test_networkevents
241.39s call     test_plugin.py::test_plugin_startdir_lol
240.74s call     test_gossip.py::test_gossip_pruning
204.50s call     test_gossip.py::test_gossip_force_broadcast_channel_msgs
149.41s call     test_gossip.py::test_gossip_query_channel_range
129.42s call     test_connection.py::test_last_stable_connection
129.14s call     test_currencyrate.py::test_bkpr_currencyrate_warns_for_old_events
122.48s call     test_plugin.py::test_plugin_command
120.75s call     test_plugin.py::test_plugin_slowinit
120.28s call     test_plugin.py::test_failing_plugins
=========================== short test summary info ============================
FAILED tests/test_plugin.py::test_inline_plugin_wait_for_log_no_selfmatch - OSError: AF_UNIX path too long
```

AF_UNIX path too long :/

--- a/.github/workflows/macos.yaml
+++ b/.github/workflows/macos.yaml
@@ -62,5 +62,6 @@ jobs:
             tests/test_misc.py::test_ipv4_and_ipv6
             tests/test_connection.py::test_websocket
             tests/test_connection.py::test_wss_proxy
+            tests/test_plugin.py::test_inline_plugin_wait_for_log_no_selfmatch
         run: |
           VALGRIND=0 uv run pytest $PYTEST_TESTS -n $(sysctl -n hw.ncpu) ${PYTEST_OPTS}

--- a/contrib/pyln-testing/pyln/testing/utils.py
+++ b/contrib/pyln-testing/pyln/testing/utils.py
@@ -12,6 +12,7 @@ from pyln.client import LightningRpc
 from pyln.client import Millisatoshi
 from pyln.client import NodeVersion
 from pyln.client import Plugin
+from pyln.client.plugin import PluginLogHandler
 
 import ephemeral_port_reserve  # type: ignore
 import tempfile
@@ -2050,6 +2051,26 @@ class NodeFactory(object):
         return not unexpected_fail, err_msgs
 
 
+class _NoPylnInternalsFilter(logging.Filter):
+    """Drop log records generated inside the pyln packages themselves.
+
+    An inline plugin's Plugin() lives in the test process, so its
+    PluginLogHandler on the root logger would forward pyln's own machinery
+    logs into the node's log.  wait_for_logs()'s 'Waiting for [pattern]'
+    announcement embeds the pattern verbatim, lands in the very log being
+    scanned, and matches itself, silently reducing the wait to a no-op.
+    Only records from outside pyln (i.e. the plugin author's own logging)
+    may be forwarded.
+    """
+    PYLN_DIRS = tuple(
+        os.path.dirname(os.path.abspath(f)) + os.sep
+        for f in (__file__,
+                  sys.modules[PluginLogHandler.__module__].__file__))
+
+    def filter(self, record):
+        return not os.path.abspath(record.pathname).startswith(self.PYLN_DIRS)
+
+
 def _inline_plugin(node, setup_fn):
     """Set up an inline plugin serve thread for a not-yet-started node.
 
@@ -2069,10 +2090,33 @@ def _inline_plugin(node, setup_fn):
     """
     sock_path = os.path.join(node.daemon.lightning_dir, TEST_NETWORK, 'inline-plugin.sock')
     srv = socket.socket(socket.AF_UNIX)
-    srv.bind(sock_path)
+    try:
+        srv.bind(sock_path)
+    except OSError as e:
+        # AF_UNIX caps the bind path (108 bytes on Linux, 104 on macOS),
+        # and the node dir embeds the (possibly long) test name.  Bind
+        # through a short symlink alias to the socket's directory -- the
+        # bind-side analogue of UnixSocket.connect's Darwin workaround
+        # (bind can't go through a dangling final-component symlink, so
+        # alias the directory rather than the socket).  The socket file
+        # still lands at sock_path, where the shim's cwd-relative connect
+        # expects it.
+        if e.args[0] != "AF_UNIX path too long":
+            raise
+        alias_dir = tempfile.mkdtemp(prefix='pyln-sock-')
+        alias = os.path.join(alias_dir, 'd')
+        os.symlink(os.path.dirname(sock_path), alias)
+        try:
+            srv.bind(os.path.join(alias, os.path.basename(sock_path)))
+        finally:
+            os.unlink(alias)
+            os.rmdir(alias_dir)
     srv.listen(1)
 
     plugin = Plugin(autopatch=False)
+    for handler in logging.getLogger().handlers:
+        if isinstance(handler, PluginLogHandler) and handler.plugin is plugin:
+            handler.addFilter(_NoPylnInternalsFilter())
     setup_fn(plugin)
 
     def serve():

--- a/tests/test_plugin.py
+++ b/tests/test_plugin.py
@@ -20,6 +20,7 @@ from tests.test_wallet import HsmTool, write_all, WAIT_TIMEOUT
 import ast
 import copy
 import json
+import logging
 import os
 import pytest
 import random
@@ -1801,6 +1802,38 @@ def test_sendpay_notifications_nowaiter(node_factory):
     results = l1.rpc.call('listsendpays_plugin')
     assert len(results['sendpay_success']) == 1
     assert len(results['sendpay_failure']) == 1
+
+
+def test_inline_plugin_wait_for_log_no_selfmatch(node_factory):
+    """On inline-plugin nodes the test process's logging is forwarded into
+    the node's log.  wait_for_log()'s own 'Waiting for [pattern]'
+    announcement embeds the pattern, so with test logging at DEBUG it used
+    to land in the scanned log and match itself, reducing the wait to a
+    no-op (#9343).  A pattern that never appears must genuinely time out.
+    """
+    def setup(plugin):
+        @plugin.method('inline_ping')
+        def inline_ping(plugin):
+            logging.info("AUTHOR_LOG_MARKER_9343")
+            return {'pong': True}
+
+    l1 = node_factory.get_node(inline_plugin=setup)
+
+    root = logging.getLogger()
+    old_level = root.level
+    root.setLevel(logging.DEBUG)
+    try:
+        # The plugin author's own logging must still be forwarded...
+        assert l1.rpc.call('inline_ping') == {'pong': True}
+        l1.daemon.wait_for_log('AUTHOR_LOG_MARKER_9343')
+        # ...but pyln's internal announcements must not be, so a pattern
+        # that never appears genuinely times out instead of matching the
+        # forwarded 'Waiting for [pattern]' line.
+        with pytest.raises(TimeoutError):
+            l1.daemon.wait_for_log('SELFMATCH_SENTINEL_NEVER_LOGGED',
+                                   timeout=5)
+    finally:
+        root.setLevel(old_level)
 
 
 def test_rpc_command_hook(node_factory):

